pandas/core/frame.py

```diff
@@ -5177,53 +5177,55 @@ def append(self, other, ignore_index=False, verify_integrity=False):
     def join(self, other, on=None, how='left', lsuffix='', rsuffix='',
              sort=False):
         """
-        Join columns with other DataFrame either on index or on a key
-        column. Efficiently Join multiple DataFrame objects by index at once by
+        Join columns with other DataFrame either on index or a key
+        column.
+
+        Efficiently Join multiple DataFrame objects by index at once by
         passing a list.
 
         Parameters
         ----------
         other : DataFrame, Series with name field set, or list of DataFrame
```

Comment:
Without trying to do to much, perhaps this line should just be `DataFrame, Series or list of DataFrame`. The detail around the name being set is in the comment below, and could maybe be reinforced in the `Notes` section